railties/test/backtrace_cleaner_test.rb

```diff
@@ -43,6 +43,13 @@ def setup
     assert_equal "app/views/application/index.html.erb:4", result[0]
   end
 
+  test "#clean should omit ActionView template methods names on Ruby 3.4+" do
+    method_name = ActionView::Template.new(nil, "app/views/application/index.html.erb", nil, locals: []).send :method_name
+    backtrace = [ "app/views/application/index.html.erb:4:in 'block in #{method_name}'"]
```

Comment:
How about just adding a ruby version conditional for the [backtrace var](https://github.com/rails/rails/blob/main/railties/test/backtrace_cleaner_test.rb#L41)? Surrounding code won't need to change and no additional tests.

Reply:
That could also be fine, but there is no harm in additional fast tests and I prefer this since the implementation also covers both versions.